src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingStructureBuilder.java

```diff
@@ -333,7 +333,7 @@ public void serializeToView(@NotNull final PacketBuffer buf)
                 }
 
                 buf.writeString(desc);
-                buf.writeDouble(workOrderBuildDecoration.getAmountOfRes() == 0 ? 0 : qty / workOrderBuildDecoration.getAmountOfRes());
+                buf.writeDouble(workOrderBuildDecoration.getAmountOfRes() == 0 ? 0 : (qty < workOrderBuildDecoration.getAmountOfRes() ? qty / workOrderBuildDecoration.getAmountOfRes() : 1.0));
```

Comment:
Were you able to reproduce this happenin?

Reply:
Yes, though not controllably enough to find the underlying reason getAmountOfRes was only partially filled.  Very large builds where the builder had to clear a lot of spaces and other modifications to the build area occurred seemed most likely to have it happen, but even with slower AI config it didn't always show up even when working the same build at the same location.

Reply:
I tried hunting this down numerous times but couldn't reproduce it. It's weird because the amoutnOfRes is only set once, and that is after the scanning step is done. And it is hooked in, that if a rescan is done, and the new qty is bigger, it will adopt the new quantity.

Reply:
It definitely can get called multiple times in weird situations.  One example case: building a townhall1 in a spruce forest biome, with a builder2, warehouse1, four colonists, build order issued before the builder had complete builder2.  First setAmountOfRes sets from 0 to 8, and only requests eight torches.  At this time, the progress value is correct(ish) at 0%.

Builder begins clearing phase, cuts out trees, ferns, and ground clutter.  When the clearing phase is complete, the builder recalculates the qty value, but does not yet issue a new setAmountOfRes().

![image](https://user-images.githubusercontent.com/39036909/110990629-8b699f00-8341-11eb-8407-0ba8b0b8976a.png)

Reply:
Raycoms traced the problem down to, if I'm understanding it correctly, requestMaterialsState() not updating setAmountOfRes when in the SOLID, DECO, or ENTITIES state, and that seems a far more elegant fix.  Removed the extra ternary, and pushed his fix instead.  At least cases where I'd replicated the problem previously seem to be reliably giving non-crazy numbers.